queries/ocaml/hightlights.scm

```diff
@@ -0,0 +1,146 @@
+(module_name) @structure
+
+[
+  (module_type_name)
+  (type_constructor)
+  (type_variable)
+  (constructor_name)
+  (tag)
+] @type
+(type_binding (type_constructor) @type)
+
+(exception_definition (constructor_declaration (constructor_name) @exception))
+(application_expression (value_path (value_name) @_raise)
+                        (#eq? @_raise "raise")
+                        (constructor_path) @exception)
+(application_expression (value_path (value_name) @_raise)
+                        (#eq? @_raise "raise")
+                        (parenthesized_expression
+                        (application_expression (constructor_path) @exception)))
+(infix_expression (value_path (value_name) @_raise)
+                  (#eq? @_raise "raise")
+                  (application_expression (constructor_path) @exception))
+
+(attribute_id) @property
+(field_name) @field
+
+(let_binding
+  (value_name) @function
+  (parameter))
+(let_binding
+  (value_name) @function
+  (function_expression))
+
+(application_expression . (value_path (value_name) @function))
```

Comment:
It looks like you set `(value_path (value_name) @function)` everywhere, but above where it's `@_raise` you can shortcut this by extracting it and having it at the top-level. The highlighter will then default it to `@function` unless a more specific query is used.

Reply:
The problem is that `(value_path (value_name))` matches any value, function or not. So I have to depend on the outermost context. I've raised the issue https://github.com/tree-sitter/tree-sitter-ocaml/issues/40 to prevent this awkwardness.